src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -226,7 +226,7 @@ private bool UpToDate
             }
 
             public void RegisterNotification(Action action, int delay, CancellationToken cancellationToken)
-                => _notificationService.RegisterNotification(action, delay, _asyncListener.BeginAsyncOperation("TagSource"), cancellationToken);
+                => _notificationService.RegisterNotification(action, delay, _asyncListener.BeginAsyncOperation(typeof(TTag).Name), cancellationToken);
```

Comment:
Nit: want to do something like typeof(...).Name + "TagSource"? That way if you have a greppable string in addition to useful detail. (If you ended up holding a token with name IOutliningTag, it's less obvious what to find?)

Reply:
:arrow_right: Name chosen to provide meaning without incurring a fast path allocation (the method is a NOP in production)